src/main/java/io/thinkit/edc/client/connector/services/VerifiableCredentials.java

```diff
@@ -0,0 +1,175 @@
+
+    public CompletableFuture<Result<String>> deleteAsync(String participantId, String credentialId) {
+        var requestBuilder = deleteRequestBuilder(participantId, credentialId);
+
+        return this.edcApiHttpClient
+                .sendAsync(requestBuilder)
+                .thenApply(result -> result.map(content -> participantId));
+    }
+
+    private HttpRequest.Builder getRequestBuilder(String participantId, String credentialId) {
+        return HttpRequest.newBuilder()
+                .uri(URI.create("%s/participants/%s/credentials/%s".formatted(this.url, participantId, credentialId)))
+                .GET();
+    }
+
+    private HttpRequest.Builder getListRequestBuilder(String participantId, String type) {
+        return HttpRequest.newBuilder()
+                .uri(URI.create("%s/participants/%s/credentials?type=%s".formatted(this.url, participantId, type)))
+                .GET();
+    }
+
+    private HttpRequest.Builder getAllRequestBuilder(int offset, int limit) {
+        return HttpRequest.newBuilder()
+                .uri(URI.create("%s/credentials?offset=%s&limit=%s".formatted(this.url, offset, limit)))
+                .GET();
+    }
+
+    private HttpRequest.Builder createRequestBuilder(VerifiableCredentialManifest input, String participantId) {
+        String requestBody = null;
+        try {
+            requestBody = objectMapper.writeValueAsString(input);
+        } catch (JsonProcessingException e) {
+            throw new RuntimeException(e);
+        }
+        return HttpRequest.newBuilder()
+                .uri(URI.create("%s/participants/%s/credentials".formatted(this.url, participantId)))
+                .header("content-type", "application/json")
+                .POST(HttpRequest.BodyPublishers.ofString(requestBody));
+    }
+
+    private HttpRequest.Builder updateRequestBuilder(VerifiableCredentialManifest input, String participantId) {
+        String requestBody = null;
+        try {
+            requestBody = objectMapper.writeValueAsString(input);
+        } catch (JsonProcessingException e) {
+            throw new RuntimeException(e);
+        }
+        return HttpRequest.newBuilder()
+                .uri(URI.create("%s/participants/%s/credentials".formatted(this.url, participantId)))
+                .header("content-type", "application/json")
+                .PUT(HttpRequest.BodyPublishers.ofString(requestBody));
+    }
+
+    private HttpRequest.Builder deleteRequestBuilder(String participantId, String credentialId) {
+        return HttpRequest.newBuilder()
+                .uri(URI.create("%s/participants/%s/credentials/%s".formatted(this.url, participantId, credentialId)))
+                .DELETE();
+    }
+
+    private VerifiableCredentialResource getVerifiableCredential(InputStream body) {
+        try {
+            return objectMapper.readValue(body, VerifiableCredentialResource.class);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
```

Comment:
it would be nice to return a failed Result here, so the error will be brought up to the caller.
this could be done in a different issue/PR, eventually please create the issue